### PR TITLE
Add Key Vault article

### PR DESCRIPTION
**Pull request by JasonFreeberg**
Adds a page for setting Key Vault references

### CODE REVIEW

File: step-06-secure-secrets-in-key-vault/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+The command wil output information about the identity. Copy the `principalId` for the next section.
+
+## Create a Key Vault
+
+Next, provision an Azure Key Vault, provide access to the identity you just created, and set the database username, password, and URL in Key Vault.
+
+1. Create the Key Vault using the Azure CLI.
+
+    ```bash
+    az keyvault create --name jboss-app-key-vault              \
```
Comment (selvasingh) [inline step-06-secure-secrets-in-key-vault/README.md]: For name, resource group and location, we should rely on environment variables. Those variables should be derived from how we have requested users to create a copy of https://github.com/Azure-Samples/migrate-javaee-app-to-azure-training/blob/main/setup-env-variables-template.sh

File: step-06-secure-secrets-in-key-vault/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+                        --enabled-for-deployment true          \
+                        --enabled-for-disk-encryption true     \
+                        --enabled-for-template-deployment true \
+                        --sku standard
+    ```
+
+1. Now grant the managed identity `get` and `list` access to the Key Vault.
+
+    ```bash
+    az keyvault set-policy --name java-app-key-vault     \
```
Comment (selvasingh) [inline step-06-secure-secrets-in-key-vault/README.md]: We should add an additional environment variable and set the object id before calling this cli command

File: step-06-secure-secrets-in-key-vault/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                            --secret-permission get list  \
+                            --object-id <the principal ID from earlier>
+    ```
+
+1. Lastly, add the Postgres username, password, and URL to the Key Vault. If you followed sections 2 and 3, you should still have the secrets saved as environment variables on your machine.
+
+    ```bash
+    az keyvault secret set --name POSTGRES_SERVER_ADMIN_FULL_NAME \
```
Comment (selvasingh) [inline step-06-secure-secrets-in-key-vault/README.md:50]: Vault name should be retrieved from an environment variable

File: step-06-secure-secrets-in-key-vault/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    ```
+
+## Set Key Vault References on App Service
+
+The secrets will be exposed as environment variables to the JBoss application, so the final step is to get the URI's of the Key Vault secrets and set them as app settings on the Web App (with the appropriate syntax).
+
+1. Get the URI’s of your three secrets. Run the commands below and copy the `id` value in the console output.
+
+    ```bash
+    az keyvault secret show --vault-name java-app-key-vault --name POSTGRES-URL
```
Comment (selvasingh) [inline step-06-secure-secrets-in-key-vault/README.md]: Vault name should be retrieved from an environment variable

File: step-06-secure-secrets-in-key-vault/README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    az keyvault secret show --vault-name java-app-key-vault --name POSTGRES-USERNAME
+    az keyvault secret show --vault-name java-app-key-vault --name POSTGRES-PASSWORD
+    ```
+
+2. Now create the app settings with the Key Vault references. For each setting, replace “YOUR_SECRET_URI” with the corresponding id’s from the previous step.
+
+    ```bash
+    az webapp config appsettings set -n <your_app_name> -g <resource_group> --settings \
```
Comment (selvasingh) [inline step-06-secure-secrets-in-key-vault/README.md]: Spring? :-)

File: step-06-secure-secrets-in-key-vault/README.md (hunk 3 above)
Comment (selvasingh) [reply step-06-secure-secrets-in-key-vault/README.md:50]: We should have 3 sub sections for each of the databases - SQL, PostgreSQL and MySQL

**Review by selvasingh — CHANGES_REQUESTED** [review]

File: step-06-secure-secrets-in-key-vault/README.md (hunk 5 above)
Comment (JasonFreeberg) [reply step-06-secure-secrets-in-key-vault/README.md]: Copied from my older article

**Pull request closed without merging**